### PR TITLE
Added strikethrough option to Font

### PR DESCRIPTION
**Pull request by MightyJosip**
This adds strikethrough option to pygame.font.Font. It was already present in SDL_ttf, but it wasn't ported to pygame.
image

### CODE REVIEW

**Comment by Starbuck5** [comment]
Tests are failing because of the ftfont module. For the unfamiliar, ftfont is a compatibility layer of the pygame Font API implemented on top of pygame.freetype.

For a long time, pygame Font was a subset of freetype. But with new advances in SDL_ttf, including line wrapping soon, we’re going to have to deal with that no longer being the case.

And we’ll have to make some decisions about ftfont and about how much effort we want to spend keeping freetype up to date with Font.

**Comment by Starbuck5** [comment]
I think the way to go forward is to make ftfont test skip these somehow, and put a tiny note that the ftfont module doesn't support underline in the docs.

**Comment by MyreMylar** [comment]
> I think the way to go forward is to make ftfont test skip these somehow, and put a tiny note that the ftfont module doesn't support strikethrough in the docs.

I found a way to skip ftfont tests in the short term for the recent sdl_ttf_version PR:

```
def test_get_sdl_ttf_version(self):
    if pygame_font.__name__ != "pygame.ftfont":
        pygame_font.get_sdl_ttf_version()
```
So adapting that for this test is probably the way to go for this PR.

I think the long term solution is just to scrap ftfont via streamlining us down to one font back end and fewer user facing options (see the linked issue).

File: test/font_test.py [hunk 1]
```diff
@@ -619,6 +647,9 @@ def test_italic(self):
     def test_underline(self):
         self.assertTrue(self.query(underline=True))
 
+    def test_strikethrough(self):
+        self.assertTrue(self.query(strikethrough=True))
```
Comment (MyreMylar) [inline test/font_test.py:651]:
```suggestion
        if pygame_font.__name__ != "pygame.ftfont":
            self.assertTrue(self.query(strikethrough=True))
```

File: test/font_test.py [hunk 2]
```diff
@@ -628,6 +659,9 @@ def test_bold_antialiase(self):
     def test_italic_underline(self):
         self.assertTrue(self.query(italic=True, underline=True))
 
+    def test_bold_strikethrough(self):
+        self.assertTrue(self.query(bold=True, strikethrough=True))
```
Comment (MyreMylar) [inline test/font_test.py:663]:
```suggestion
        if pygame_font.__name__ != "pygame.ftfont":
            self.assertTrue(self.query(bold=True, strikethrough=True))
```

File: test/font_test.py [hunk 3]
```diff
@@ -581,18 +605,22 @@ def query(self, bold=False, italic=False, underline=False, antialiase=False):
...
         f.set_italic(italic)
         f.set_underline(underline)
+        f.set_strikethrough(strikethrough)
         s = f.render(text, antialiase, (0, 0, 0))
         screen.blit(s, (offset, y))
         y += s.get_size()[1] + spacing
         f.set_bold(False)
         f.set_italic(False)
         f.set_underline(False)
+        f.set_strikethrough(False)
```
Comment (MyreMylar) [inline test/font_test.py:623]:
```suggestion
        if pygame_font.__name__ != "pygame.ftfont":        
            f.set_strikethrough(False)
```

File: test/font_test.py [hunk 4]
```diff
@@ -581,18 +605,22 @@ def query(self, bold=False, italic=False, underline=False, antialiase=False):
...
                 modes.append("underlined")
+            if strikethrough:
+                modes.append("strikethrough")
             if antialiase:
                 modes.append("antialiased")
             text = "%s (y/n):" % ("-".join(modes),)
         f.set_bold(bold)
         f.set_italic(italic)
         f.set_underline(underline)
+        f.set_strikethrough(strikethrough)
```
Comment (MyreMylar) [inline test/font_test.py:616]:
```suggestion
        if pygame_font.__name__ != "pygame.ftfont":
            f.set_strikethrough(strikethrough)
```

File: test/font_test.py [hunk 5]
```diff
@@ -422,6 +423,14 @@ def test_set_underline(self):
         f.set_underline(False)
         self.assertFalse(f.get_underline())
 
+    def test_set_strikethrough(self):
+        f = pygame_font.Font(None, 20)
+        self.assertFalse(f.get_strikethrough())
+        f.set_strikethrough(True)
+        self.assertTrue(f.get_strikethrough())
+        f.set_strikethrough(False)
+        self.assertFalse(f.get_strikethrough())
```
Comment (MyreMylar) [inline test/font_test.py:432]:
```suggestion
        if pygame_font.__name__ != "pygame.ftfont":
            f = pygame_font.Font(None, 20)
            self.assertFalse(f.get_strikethrough())
            f.set_strikethrough(True)
            self.assertTrue(f.get_strikethrough())
            f.set_strikethrough(False)
            self.assertFalse(f.get_strikethrough())
```

File: test/font_test.py [hunk 6]
```diff
@@ -446,6 +455,14 @@ def test_set_underline_property(self):
         f.underline = False
         self.assertFalse(f.underline)
 
+    def test_set_strikethrough_property(self):
+        f = pygame_font.Font(None, 20)
+        self.assertFalse(f.strikethrough)
+        f.strikethrough = True
+        self.assertTrue(f.strikethrough)
+        f.strikethrough = False
+        self.assertFalse(f.strikethrough)
```
Comment (MyreMylar) [inline test/font_test.py:464]:
```suggestion
        if pygame_font.__name__ != "pygame.ftfont":
            f = pygame_font.Font(None, 20)
            self.assertFalse(f.strikethrough)
            f.strikethrough = True
            self.assertTrue(f.strikethrough)
            f.strikethrough = False
            self.assertFalse(f.strikethrough)
```

**Review by MyreMylar — COMMENTED** [review]
batching up some test fixes to help out.

**Comment by MyreMylar** [comment]
Hmm, not sure I can run the docs generation via the GitHub interface to finish fixing up this PR

Any ideas?

File: docs/reST/ref/font.rst [hunk 7]
```diff
@@ -309,6 +327,30 @@ solves no longer exists, it will likely be removed in the future.
...
+   .. method:: get_strikethrough
+
+      | :sl:`check if text will be rendered with a strikethrough`
+      | :sg:`get_strikethrough() -> bool`
+
+      Return True when the font strikethrough is enabled.
+
+       .. note:: This is the same as the :attr:`strikethrough` attribute.
+
+      .. ## Font.get_strikethrough ##
```
Comment (ankith26) [inline docs/reST/ref/font.rst:353]: Both above function docs probably need `.. versionadded:: 2.1.3`

**Review by ankith26 — APPROVED** [review]
Nice! Thanks for the PR 🥳 🎉 

(left a minor review)

**Review by MyreMylar — APPROVED** [review]
LGTM 👍

File: test/font_test.py [hunk 8]
```diff
@@ -1,3 +1,5 @@
+# -*- coding: utf-8 -*-
+from re import T
```
Comment (illume) [inline test/font_test.py:2]: I wonder what this import was added for? I can't see this T variable used anywhere.

File: docs/reST/ref/font.rst [hunk 9]
```diff
@@ -211,11 +212,28 @@ solves no longer exists, it will likely be removed in the future.
...
+   
+   .. attribute:: strikethrough
+
+      | :sl:`Gets or sets whether the font should be rendered with a strikethrough.`
+      | :sg:`strikethrough -> bool`
+
+      Whether the font should be rendered with a strikethrough.
+
+      When set to True, all rendered fonts will include an
+      strikethrough. The strikethrough is always one pixel thick,
```
Comment (illume) [inline docs/reST/ref/font.rst:230]: "include an strikethrough" should be "include a strikethrough"

**Pull request merged**